Content.Server/Body/Behavior/StomachBehavior.cs

```diff
@@ -61,14 +65,27 @@ public override void Update(float frameTime)
                 delta.Increment(1);
                 if (delta.Lifetime > _digestionDelay)
                 {
-                    solution.TryRemoveReagent(delta.ReagentId, delta.Quantity);
-                    transferSolution.AddReagent(delta.ReagentId, delta.Quantity);
+                    // This reagant has been in the somach long enough, TRY to transfer it.
+                    // But first, check if the reagant still exists, and how much is left.
+                    // Some poort spessman may have washed down a potassium snack with some water.
+                    if (solution.Solution.ContainsReagent(delta.ReagentId, out ReagentUnit quantity)){
+
+                        if (quantity > delta.Quantity) {
+                            quantity = delta.Quantity;
+                        }
+
+                        solution.TryRemoveReagent(delta.ReagentId, quantity);
+                        transferSolution.AddReagent(delta.ReagentId, quantity);
+                    }
+
                     _reagentDeltas.Remove(delta);
                 }
             }
 
-            // Transfer digested reagents to bloodstream
-            bloodstream.TryTransferSolution(transferSolution);
+            // Transfer digested reagents to bloodstream, if there is something to actually transfer
+            if (transferSolution.TotalVolume > 0){ 
+                bloodstream.TryTransferSolution(transferSolution);
+            }
```

Comment:
this check should be the job of TryTransferSolution to handle imo

Reply:
Ah yes, it already is. This was accidentally left in. It was a remant from when I was debugging bloodstream and wanted to minimize the number of calls to bloodstream.TryTransferSolution(), as otherwise StomachBehaviour calls it once every second.  I will revert it.